### PR TITLE
Create and style the time frame overlay

### PR DESCRIPTION
**Pull request by tramtran0497**

### CODE REVIEW

File: src/layout/components/CalendarDay/TimeFrameOverlay.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+type TimeFrameOverlayProps = {
+    title: string,
+    content: string,
+    bgColor?: string,
+    top: number,
+    height: number,
+}
+
+const TimeFrameOverlay = ({title, content, bgColor, top, height}: TimeFrameOverlayProps) => {
+    const numberOfLines = Math.round((height - 24)/24)
```
Comment (LeTranAnhVu) [inline src/layout/components/CalendarDay/TimeFrameOverlay.tsx:12]: - What is 24? 
- If the logic is not clear, please add some comment above it.
- I thin we can use `useMemo` in this case

File: src/layout/components/CalendarDay/TimeFrameOverlay.scss [hunk 2]
```diff
@@ -0,0 +1,57 @@
+.time-frame-overlay-wrapper {
```
Comment (LeTranAnhVu) [inline src/layout/components/CalendarDay/TimeFrameOverlay.scss:1]: Clean all unused comment codes in this file and also check all changed files the PR

File: src/lib/__tests__/timeToHeightPx.test.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import { durationToHeightPx } from '../helpers/durationToHeightPx';
+
+describe('durationToHeightPx', () => {
+    it('Throw error when hour is less than 0', () => {
+      expect(() => durationToHeightPx(-1, 47)).toThrow()
+    })
+
+    it('Throw error when hour is equal 0', () => {
+        expect(() => durationToHeightPx(0, 47)).toThrow()
+    })
```
Comment (LeTranAnhVu) [inline src/lib/__tests__/timeToHeightPx.test.ts:11]: You can make it shorter in this case by using `test.each` . Check out the example in here : https://dev.to/bgord/simplify-repetitive-jest-test-cases-with-test-each-310m

File: src/layout/components/CalendarDay/TimeFrameOverlay.scss [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        height: 100%;
+        position: absolute;
+        z-index: 0;
+        opacity: 0.15;
+    }
+    position: relative;
+    z-index: 1;
+
+    width: 100%; // it can be modify
+    height: 47px; // default height for 1 hour
```
Comment (LeTranAnhVu) [inline src/layout/components/CalendarDay/TimeFrameOverlay.scss:15]:
```suggestion
```
Useless code, remove it

File: src/layout/components/CalendarDay/TimeFrameOverlay.scss [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    margin: 1px 0px;
```
Comment (LeTranAnhVu) [inline src/layout/components/CalendarDay/TimeFrameOverlay.scss:16]:
```suggestion
```

File: src/layout/components/CalendarDay/TimeFrameOverlay.scss [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    margin: 1px 0px; 
+    border-radius: 3px;
+    border-left: solid 3px $blue-light;
+    overflow: hidden;
+    
+    .time-frame-overlay-title {
+        position: relative;
+        z-index: 1;
+
+        display: -webkit-box;
```
Comment (LeTranAnhVu) [inline src/layout/components/CalendarDay/TimeFrameOverlay.scss:25]: Should add some comment to explain the style, better for reviewing it later

File: src/layout/components/CalendarDay/TimeFrameOverlay.scss [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        -webkit-box-orient: vertical;
+        // white-space: wrap;
+        overflow: hidden;
+        // display: inline-block;
+        text-overflow: ellipsis;
+
+        margin-left: 2px;
+        color: $blue-light; // default color
+        font-size: 11px;
```
Comment (LeTranAnhVu) [inline src/layout/components/CalendarDay/TimeFrameOverlay.scss:55]: Use `em`

Comment (LeTranAnhVu) [reply src/layout/components/CalendarDay/TimeFrameOverlay.scss:55]: Around 0.8 or 0.9 `em`

**Review by LeTranAnhVu — CHANGES_REQUESTED** [review]
- Update main branch to this PR
- Check eslint and prettier

File: src/layout/components/CalendarDay/TimeFrameOverlay.tsx (hunk 1 above)
Comment (tramtran0497) [reply src/layout/components/CalendarDay/TimeFrameOverlay.tsx]: - 24 is line-height

**Pull request merged**